pkg/stdlib/objects/merge.go

```diff
@@ -0,0 +1,67 @@
+package objects
+
+import (
+	"context"
+
+	"github.com/MontFerret/ferret/pkg/runtime/core"
+	"github.com/MontFerret/ferret/pkg/runtime/values"
+)
+
+/*
+ * Merge the given objects into a single object.
+ * @params objs (Array Of Object OR Objects) - objects to merge.
+ * @returns (Object) - Object created by merging.
+ */
+func Merge(_ context.Context, args ...core.Value) (core.Value, error) {
+	err := core.ValidateArgs(args, 1, core.MaxArgs)
+
+	if err != nil {
+		return values.None, err
+	}
+
+	objs := values.NewArrayWith(args...)
+
+	if len(args) == 1 && args[0].Type() == core.ArrayType {
+		objs = args[0].(*values.Array)
+	}
+
+	err = validateArrayOf(core.ObjectType, objs)
+
+	if err != nil {
+		return values.None, err
+	}
+
+	return mergeArray(objs), nil
+}
+
+func mergeArray(arr *values.Array) *values.Object {
+	merged := values.NewObject()
+
+	var val core.Value
+	var k values.String
+	var obj *values.Object
+
+	for idx := values.NewInt(0); idx < arr.Length(); idx++ {
+		obj = arr.Get(idx).(*values.Object)
+		for _, key := range obj.Keys() {
+			k = values.NewString(key)
+			val, _ = obj.Get(k)
+			if values.IsCloneable(val) {
+				val = val.(core.Cloneable).Clone()
```

Comment:
Do you think we really need to clone it? I was thinking of just copying.
Could you elaborate on that?

Reply:
Variables in AQL are immutable once assigned. So, if we simply apply ```Copy```, then we will break this principle and we will be able to change the nested objects from the outside.

Reply:
So, do I continue to use the ```Clone``` function?

Reply:
Yep.